src/pubtools/_marketplacesvm/tasks/community_push/command.py

```diff
@@ -443,58 +442,68 @@ def _check_product_in_rhsm(self, enriched_push_items: List[EnrichedPushItem]) ->
+        )
 
-            # Prepare the sharing accounts
-            sharing_accts: SharingAccounts = push_items_and_sa.sharing_accounts
-            additional_args = {}
-            extra_args = ["accounts", "sharing_accounts", "snapshot_accounts"]
-            for arg in extra_args:
-                content = sharing_accts.get(arg)
-                if content:
-                    additional_args[arg] = content
-
-            # Upload the push items in parallel
-            log.info("Uploading to the storage account %s", storage_account)
-            for pi in push_items:
-                to_await.append(
-                    executor.submit(self._upload, storage_account, pi, **additional_args)
-                )
+        # consume the queue
+        for data in push_queue:
+            to_await.append(executor.submit(self._upload, **data))
 
-            # Wait for all results
-            for f_out in to_await:
-                out_pi.append(f_out.result())
-
-            # Append the data for collection
-            for pi in out_pi:
-                result.append(
-                    {
-                        "push_item": pi,
-                        "state": pi.state,
-                        "image_id": pi.image_id,
-                        "image_name": name_from_push_item(pi),
-                    }
-                )
-        return result
+        # wait for results
+        for f_out in to_await:
+            upload_result.append(f_out.result())
+
+        # Return the data for collection
+        return [
+            {
+                "push_item": pi,
+                "state": pi.state,
+                "image_id": pi.image_id,
+                "image_name": name_from_push_item(pi),
+            }
+            for pi in upload_result
+        ]
+
+    def _data_to_upload(
+        self, enriched_push_items: List[EnrichedPushItem]
+    ) -> Iterator[Dict[str, Any]]:
+        """
+        Generate the required data to perform the AMI upload and update RHSM.
+
+        Args:
+            enriched_push_items
+                List of dictionaries and storage accounts with the region name and the push items.
+        """
+        for enriched_push_item in enriched_push_items:
+            for storage_account, push_items_and_sa in enriched_push_item.items():
+                push_items = push_items_and_sa.push_items
+
+                # Prepare the sharing accounts
+                sharing_accts: SharingAccounts = push_items_and_sa.sharing_accounts
+                additional_args = {}
+                extra_args = ["accounts", "sharing_accounts", "snapshot_accounts"]
+                for arg in extra_args:
+                    content = sharing_accts.get(arg)
+                    if content:
+                        additional_args[arg] = content
+
+                # Generate the push items to upload
+                for pi in push_items:
+                    yield {"marketplace": storage_account, "push_item": pi, **additional_args}
```

Comment:
@lslebodn the log message `Uploading to the storage account ACCOUNT` was originally here but it don't make sense to set this in the producer and by adding that in the consumer would create a lot of repetitive messages and turn everything into an overcomplication so I just removed it (that's why the tests logs were updated).